packages/astro/src/core/messages.ts

```diff
@@ -116,6 +114,37 @@ export function devStart({
 	return messages.map((msg) => `  ${msg}`).join('\n');
 }
 
+export function telemetryNotice() {
+	const headline = yellow(`Astro now collects ${bold('anonymous')} usage data.`);
+	const why = `This ${bold('optional program')} will help shape our roadmap.`;
+	const more = `For more info, visit ${underline('https://astro.build/telemetry')}`;
+	const box = boxen([headline, why, '', more].join('\n'), {
+		margin: 0,
+		padding: 1,
+		borderStyle: 'round',
+		borderColor: 'yellow',
+	});
+	return box;
+}
+
+export function telemetryEnabled() {
+	return `\n  ${green('◉')} Telemetry is ${bgGreen(
+		black(' enabled ')
+	)}. Thank you for improving Astro!\n`;
+}
+
+export function telemetryDisabled() {
+	return `\n  ${yellow('◯')} Telemetry is ${bgYellow(
+		black(' disabled ')
+	)}. We won't share any usage data.\n`;
+}
+
+export function telemetryReset() {
+	return `\n  ${cyan('◆')} Telemetry has been ${bgCyan(
```

Comment:
nit: "Telemetry settings" or "Telemetry preference" is really what is being reset here, not telemetry itself.